src/Model/Ticket/Request/CreateMessage.php

```diff
@@ -245,20 +246,39 @@ public function setSendOperatorsEmail(?bool $sendOperatorsEmail): self
     }
 
     /**
-     * @return string[]|null
+     * @return array<string[]>|null
      */
     public function getAttachment(): ?array
     {
         return $this->attachment;
     }
 
     /**
-     * @param string[]|null $attachment
+     * This method expects an array of attachments in the following format.
+     * ['filename' => 'test', 'contents' => 'some-base-64-string'].
+     * This method overwrites any attachments you previously set.
+     * @param array<string[]>|null $attachments
      * @return self
+     * @throws InvalidArgumentException
      */
-    public function setAttachment(?array $attachment): self
+    public function setAttachment(?array $attachments): self
+    {
+        if ($attachments !== null) {
+            foreach ($attachments as $attachment) {
+                if (! isset($attachment['filename']) || ! isset($attachment['contents'])) {
+                    throw new InvalidArgumentException('Each Attachment value must include a file name, and contents.');
```

Comment:
Doesn't this break BC? `setAttachment(['filename' => 'x', 'contents' => 'y']);` will now throw an exception

Reply:
yeah.. I guess so. could just leave it into a future release then.

Reply:
Something like the below would cover what you were trying to do I think:
```php
if (isset($attachments['filename']) && isset($attachments['contents'])) {
        
} else {
  foreach ($attachments ?? [] as $attachment) {
    if (! isset($attachment['filename']) || ! isset($attachment['contents'])) {
      throw new InvalidArgumentException;
    }
  }
}
```

Reply:
don't feel that this would add a lot of robustness. I think having the docs should be OK for now, since the main issue was the vagueness of how the parameter should be passed.

Reply:
Correct me if I'm wrong but my understanding is that the Wiki (https://github.com/supportpal/api-client-php/wiki) is wrong; so we just need to update that and this PR is not necessary.

Reply:
The wiki is not wrong, it just doesn't mention the format of the attachment. In its current state, the PR is merely adding documentation on the method level.

Reply:
:ok_hand: closing